### PR TITLE
feat: group posts with any layout in archives page

### PR DESCRIPTION
**Pull request by diamantelist**
With this change, it's possible to define how to group posts in "content/post/_index.md" with the params "group_by_date".

Options:
group_by_date: "2006" (by years. Actually default)
group_by_date: "2006-01" (by months)
group_by_date: "2006-01-01" (by days)

### CODE REVIEW

**Comment by diamantelist** [comment]
Just added a commit since Decap CMS was not working with your config. Git-gateway no longer use master branch but main. I fixed.

File: layouts/_default/single.html [hunk 1]
```diff
@@ -53,7 +53,9 @@ <h1 itemprop="headline">{{ .Title }}</h1>
             {{ .Date | time.Format "Monday, Jan 2, 2006" }}
           {{ end }}
 
+          {{ if ne .Params.noreadingtime true }}
```
Comment (g1eny0ung) [inline layouts/_default/single.html:56]: Please do not introduce unrelated changes.

File: decapCMS/config.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 backend:
   name: git-gateway
-  branch: master
+  branch: main
```
Comment (g1eny0ung) [inline decapCMS/config.yml:3]: You can specify whatever branch that you used. This is an example.

File: layouts/section/posts.html [hunk 3]
```diff
@@ -16,7 +16,8 @@ <h1 class="text-3xl">
 
   <div class="max-w-[65ch] mx-auto mt-8 lg:mt-12">
     <ul class="timeline timeline-vertical">
-      {{ range .Pages.GroupByDate "2006" }}
+      {{ $groupByDate := default "2006" .Params.group_by_date }}
```
Comment (g1eny0ung) [inline layouts/section/posts.html:19]: I prefer `groupByDate` to `group_by_date` because Hugo used the camelCase format in the front matter.

Comment (g1eny0ung) [reply layouts/section/posts.html:19]: Also, refer to https://gohugo.io/methods/pages/groupbydate/#layout-string, I think it's better to name as `groupLayout`?

**Review by g1eny0ung — CHANGES_REQUESTED** [review]

File: decapCMS/config.yml (hunk 2 above)
Comment (diamantelist) [reply decapCMS/config.yml:3]: Why "I used"?! It should coincide with the branch that git-gateway dev used, not me.

Comment (g1eny0ung) [reply decapCMS/config.yml:3]: I mean it's ok to choose any branch, you can refer to this guide: https://decapcms.org/docs/backends-overview/. The `master` is just a default branch I copied from a config template, without any further changes.
